x-pack/plugins/security_solution/public/detections/components/endpoint_responder/responder_context_menu_item.tsx

```diff
@@ -62,24 +62,18 @@ export const ResponderContextMenuItem = memo<ResponderContextMenuItemProps>(
         return [true, LOADING_ENDPOINT_DATA_TOOLTIP];
       }
 
-      // if we got an error and it's a 404 (alerts can exist for endpoint that are no longer around)
+      // if we got an error and it's a 400 (alerts can exist for endpoint that are no longer around)
       // or,
       // the Host status is `unenrolled`
       if (
-        (error && error.body?.statusCode === 404) ||
+        (error && error.body?.statusCode === 400) ||
```

Comment:
I still think there is a need to check for 404 - example: what happens when you actually do call the API with an id that does not exist? I think the 400 is happening in the API after we find the record in the metadata unified index, but when we check the agent status, its unenrolled, but if we are not able to find the host id in the metadata unified index, then we return a 404 ([see here](https://github.com/elastic/kibana/blob/6efef0496077f4e61d49e4e43f75941ff3d98d9e/x-pack/plugins/security_solution/server/endpoint/services/metadata/endpoint_metadata_service.ts#L181-L185)).
Also - `400` can mean more than just the host is unenrolled. Its any user input error to the server.

Stepping back and the at this code here:

We actually should not even be checking for `statusCode` - if we got an API error, then we have no way to determine if the menu. item should be shown or not, so we should just display the unenrolled tooltip or some other message.  So this piece of code here should just check for error:

```
if (error || endpointHostInfo?.host_status === HostStatus.UNENROLLED) { //....
```